flink-table-store-core/src/main/java/org/apache/flink/table/store/CoreOptions.java

```diff
@@ -474,6 +474,20 @@ public class CoreOptions implements Serializable {
                                                             + "$hour:00:00'."))
                                     .build());
 
+    public static final ConfigOption<String> PRIMARY_KEY =
+            key("primary-key")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Define primary key by table options, cannot define primary key on DDL and table options at the same time.");
```

Comment:
Which case could the user define primary key on DDL and table options at the same time? IMO, you need to explain the case clearly.

Reply:
For Spark, the syntax for specifying partitions and primary keys is the same in `crate table` and `create table as select`.

But for Flink, `create table as select` cannot specify the partition and primary key . So we use the option parameter to specify it. However, if the user writes a following sql 
```
CREATE TABLE MyTable (
    user_id BIGINT,
    item_id BIGINT,
    behavior STRING,
    dt STRING,
    hh STRING,
    PRIMARY KEY (dt, hh, user_id) NOT ENFORCED
) WITH ('primary-key' = 'dt' );

```
we cannot determine which primary key to use. Therefore, we add verification to prevent the primary key and partition from appearing at the same time in ddl and option.